docs/java-rest/high-level/migration.asciidoc

```diff
@@ -0,0 +1,353 @@
+[[java-rest-high-level-migration]]
+== Migration Guide
+
+This section describes how to migrate existing code from the `TransportClient`
+to the new Java High Level REST Client released with the version 5.6.0
+of Elasticsearch.
+
+=== Motivations around a new Java client
+
+The existing `TransportClient` has been part of Elasticsearch since https://github.com/elastic/elasticsearch/blob/b3337c312765e51cec7bde5883bbc0a08f56fb65/modules/elasticsearch/src/main/java/org/elasticsearch/client/transport/TransportClient.java[its very first commit].
+Although it covers the whole set of Elasticsearch APIs, it always has had various flaws:
```

Comment:
> This may make people think the the high level client doesn't depend on Elasticsearch,

Right, this can be a bit confusing and it's hard to resume all the motivations in a single paragraph.

I just borrow your test and let's go!